### ISSUE TITLE
[Due for payment TBC ] [$500] Mentions styling doesn't render properly on native devices

### ISSUE THREAD

**Issue by puneetlath** [Expensify/App#19299]
On native devices (iOS and Android) the padding and border-radius on the mentions background doesn't render. This is a react native issue with nested Text elements. 

<img width="552" alt="image" src="https://github.com/Expensify/App/assets/1371996/fa469e66-6816-40ac-8e9c-afb20ef28abd">

This is being addressed in: https://github.com/Expensify/App/issues/4733

Once that is done, we should be able to make styling work on native devices. 

Issue reporter: @parasharrajat 

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021999161372801951378</li>
        <li>Upwork Job ID: 1999161372801951378</li>
        <li>Last Price Increase: 2025-12-11</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @parasharrajat</details>

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Not sure why melvin is going crazy with the overdue label.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold.

**Comment by puneetlath** [comment]
Still on hold but getting close to un-hold.

**Comment by puneetlath** [comment]
@cdOut I'm curious what you think about this one. Is it similar to https://github.com/Expensify/App/issues/27913?

**Comment by puneetlath** [comment]
Looking for assignee here: https://expensify.slack.com/archives/C04878MDF34/p1736352857632809?thread_ts=1734116044.243149&cid=C04878MDF34

**Comment by puneetlath** [comment]
Pinged SWM about it here: https://expensify.slack.com/archives/C08JJ67N602/p1746546441593819

**Comment by puneetlath** [comment]
Bumped the Slack thread.

**Comment by Skalakid** [comment]
Hello @puneetlath, @BartoszGrajdek is no longer working on the Expensify project. I will share this task with SWM team and find someone who can take over it

**Comment by Skalakid** [comment]
I'm currently finishing Bartek's inline code and code block [PR](https://github.com/Expensify/react-native-live-markdown/pull/456) in the Live Markdown library, which enables the addition of `border-radius` to the markdown style. After finishing it, I or someone from SWM can continue it and fix it also in Expensify

**Comment by puneetlath** [comment]
Ok nice, sounds great.

**Comment by puneetlath** [comment]
This still needs to happen at some point. @Skalakid is it possible now?

**Comment by war-in** [comment]
Hi 👋 I'm Marcin from SWM and I'm going to start working on this issue as soon as I finish my current issues (most important one is SignInPage and its [followups](https://github.com/Expensify/App/issues/64348))

I think I should start working on this issue this week 🤞

**Comment by war-in** [comment]
Little update here: we had a pretty hard SignInPage merge and had to resolve a few blockers and finally revert the PR.
So I'll start working on this issue once the SignInPage PR is handled properly (next week probably)

**Comment by war-in** [comment]
@parasharrajat done!

<img width="310" height="93" alt="Image" src="https://github.com/user-attachments/assets/b880d68d-b7ef-4d5e-ad71-37709d8f9aed" />

However, can't be 100% happy with the current solution as it only works when the mention is in its own paragraph. When I add some other text to it, the background will not be drawn or the background will be drawn behind this text too. It's connected to how TextKit 2 works and I need to dig deeper into the documentation to find the right solution.
Will keep you informed!

**Comment by war-in** [comment]
I managed to fix most of the android & iOS issues and the [PR](https://github.com/Expensify/react-native-live-markdown/pull/720) is looking good

However, we still have one bug which is related to the well-known single line blockquote [issue](https://github.com/Expensify/react-native-live-markdown/issues/87)
I'll need more time to investigate it and will update you with the solution 🫡

**Comment by war-in** [comment]
Together with @tomekzaw we spend some time on investigating ways of fixing the iOS singleline issue
Unfortunately, it looks like Apple doesn't expose advanced methods for drawing on the UITextField and we're still trying to find the best solution

So the current state of the PR is
- android rounded mentions work fine in both single and multi line text inputs (Composer and search input will both have rounded highlights)
- ios
  - multiline input (composer) works great 
  - singleline input (e.g. search input) still have sharp corners

Do you guys think it's fine to merge the PR in the current state and maybe set `border-radius: 0` for Android and iOS to keep the experience unchanged but have it already merged and continue working on the iOS singleline issue?
Or maybe we should wait until the right solution is found and merge the complete feature?

**Comment by war-in** [comment]
Also I and @tomekzaw are going to be OOO next week. We will contunue working on it after the holidays

**Comment by war-in** [comment]
Yes, I totally agree. I'd just put it in a separate PR at least, or even an issue

**Comment by puneetlath** [comment]
@parasharrajat can you give it a review first?

**Comment by parasharrajat** [comment]
Sure, I will take a look.

**Comment by war-in** [comment]
@parasharrajat @puneetlath I prepared an [E/App PR](https://github.com/Expensify/App/pull/76733) upgrading the `react-native-live-mardkdown` to the latest version, containing the rounded corners for mentions! (which has been released yesterday)

Unfortunately, I misplaced the issue link and deetergp was assigned, sorry!

**Comment by parasharrajat** [comment]
Payment requested as per https://github.com/Expensify/App/issues/19299#issuecomment-3721487232
